challenge7.js

```diff
@@ -0,0 +1,142 @@
+const warehouse = {
+    products: [
+        { id: 1, name: "Shoes", count: 10, price: 120, enabled: true },
+        { id: 2, name: "Socks", count: 5, price: 15, enabled: true },
+        { id: 3, name: "Jeans", count: 0, price: 320, enabled: false },
+        { id: 4, name: "Coat", count: 36, price: 456, enabled: false },
+        { id: 5, name: "Cap", count: 25, price: 64, enabled: true },
+    ],
+    soldProducts: [],
+};
+function showAllProducts(products) {
+    for (const item of products) {
+        console.log(item);
+    }
+}
+function getTotalCount() {
+    console.log(`The total count in warehouse is ${ warehouse.products.reduce(
+        (totalCount, product) => totalCount + product.count,0)
```

Comment:
Could you please explain the `accumulator`'s functionality here on the reduce method?

Reply:
Yes
The reduce method is used to add up the count of all products in the warehouse.products array. The totalCount parameter is the accumulator  and it is initialized to 0 On each iteration  the current product's count is added to the accumulator and the updated accumulator is returned.